portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -34,16 +36,45 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  // Maximum comments that can be requested
+  private static final int COMMENTS_REQUEST_LIMIT = 10000;
+
+  /** 
+   * Method that handles the GET requests to "/data" path
+   * Parameter "max-comments" specifies the maximum number of comments to return
+   * Returns a JSON array of comments ordered by addedDate descending
+   */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the request
+    String maxCommentsString = request.getParameter("max-comments");
+
+    // Convert the input to an int or Integer.MAX_VALUE
+    int maxComments;
+    try {
+      maxComments = Integer.parseInt(maxCommentsString);
+    } catch (NumberFormatException e) {
+      maxComments = Integer.MAX_VALUE;
+    }
+
+    // If the request exceeds the comments limit, bring it down
+    if(maxComments > COMMENTS_REQUEST_LIMIT) {
+        maxComments = COMMENTS_REQUEST_LIMIT;
+    }
+
+    // Make sure maxComments is not negative
+    if(maxComments < 0) {
+        maxComments = 0;
```

Comment:
Here and in line 62, do we want to throw errors instead of handling them like this?

maxComments not belonging to [0, COMMENTS_REQUEST_LIMIT] isn't intended.